library/cipher_wrap.c

```diff
@@ -566,14 +566,14 @@ static const mbedtls_cipher_info_t aes_256_xts_info = {
 #endif /* MBEDTLS_CIPHER_MODE_XTS */
 #endif /* MBEDTLS_AES_C */
 
-#if defined(MBEDTLS_GCM_C) && defined(MBEDTLS_AES_C)
+#if defined(MBEDTLS_GCM_C)
```

Comment:
Shouldn't we keep something like `MBEDTLS_CCM_GCM_CAN_AES` though?

Reply:
Previous to this commit I got undefined symbol for `ccm_aes_setkey_wrap` while trying to run `test_full_block_cipher_psa_dispatch`. This is due to the fact that the `ccm_aes_info` structure was included in that build, but `ccm_aes_setkey_wrap` was not due to `MBEDTLS_AES_C` being disabled. 
Btw I agree that I might have simplified guards too much (indeed the CI complains) so I will improve that.

Reply:
> Shouldn't we keep something like `MBEDTLS_CCM_GCM_CAN_AES` though?

Yes, it seems to work (see [commit](https://github.com/Mbed-TLS/mbedtls/pull/8632/commits/c1c6858bfc9a2e7a9c3062fa12e0acdb6d56fff1)) but it seems to me that it adds interdependency between cipher and block_cipher. Let's see what the CI thinks about it.